nemo_skills/inference/check_contamination.py

```diff
@@ -100,55 +105,118 @@ def log_example_prompt(self, data):
+        query_data_batch = [
+            query_point 
+            for data_point in data_points 
+            for query_point in self._create_query_data(data_point)
+        ]
+        
+        # Get the LLM judgement on the queries
+        outputs = super().llm_generate(query_data_batch, data, is_async)
+        
+        # Postprocessing of outputs to create a N:1 mapping between contamination results and data points
+        query_per_data_point = self.cfg.top_k * (2 if self.cfg.check_both_ways else 1)
+        
+        if not is_async:
+            output_idx = 0
+            proc_outputs = []
+            for _ in data_points:
+                proc_output = {"all_generations": [], "generation": False}
+                for output in outputs[output_idx : output_idx + query_per_data_point]:
+                    proc_output["all_generations"].append(output['generation'])
+                    # If any of the generations is True, then the data point is considered contaminated
+                    if output['generation'].strip() == "True":
+                        proc_output["generation"] = True
+                        
+                proc_outputs.append(proc_output)
+                output_idx += query_per_data_point
+
+            return proc_outputs
+        else:
+            # Club the generation IDs into tuples for each data point
+            generation_ids = []
+            for idx in range(0, len(outputs), query_per_data_point):
+                # We mark each generation ID as "Unfinished" to indicate that it is not done yet
+                generation_ids.append(outputs[idx: idx + query_per_data_point])
+
+            return generation_ids
+
+    def get_llm_generations(self, requests_in_progress, generations):
+        """Override the base class method to synchronize the N:1 mapping between contamination results and original data points. This is done by getting the LLM generations for all the queries for each data point and then processing the "generation" key.
+        
+        requests_in_progress: A dictionary of the form {original_data_point_idx: gen_id_list}
+        generations: A dictionary of the form {original_data_point_idx: gen_dict}
+        """
+        for original_dp_idx, gen_id_list in requests_in_progress.items():
+            # Get the LLM generations for all the queries remaining for this data point
+            gen_dict_list = self.llm.get_generations(gen_id_list)
+            # If all generations have finished, we can process the "generation" key, otherwise keep it as None
+            rem_gen_id_list = []
+            for gen_id, gen_dict in zip(gen_id_list, gen_dict_list):
+                if gen_dict['generation'] is None:
+                    # If any of the generations is not done, we cannot process the data point
+                    rem_gen_id_list.append(gen_id)
+                else:
+                    if "all_generations" not in generations[original_dp_idx]:
```

Comment:
please add a comment saying that this is ok to add partial generations here since we will add remaining next time this is called. Wasn't obvious to me when I first saw this code

Reply:
Cool! Added a few more comments.